public/components/event_analytics/explorer/sidebar/sidebar.tsx

```diff
@@ -55,79 +56,74 @@ export const Sidebar = (props: ISidebarProps) => {
   const [showFields, setShowFields] = useState<boolean>(false);
   const [searchTerm, setSearchTerm] = useState<string>('');
 
-  // method to return the type of a field from its name
-  const getFieldTypes = (newFieldName: string) => {
-    let fieldType: string = '';
-    explorerFields.availableFields.map((field) => {
-      if (field.name === newFieldName) fieldType = field.type;
-    });
-    explorerFields.selectedFields.map((field) => {
-      if (field.name === newFieldName) fieldType = field.type;
-    });
-    return fieldType;
-  };
-
   /**
    * Toggle fields between selected and unselected sets
    * @param fieldState all fields in store
    * @param field field to be toggled
    * @param FieldSetToRemove the set where this field to be removed from
    * @param FieldSetToAdd the set where this field to be added to
-   * returns new fields state
+   * @param indexPos position in which field should be added, default value is 0
+   * @returns new fields state
    */
   const toggleFields = (
     fieldState: ExplorerFields,
     field: IField,
     fieldSetToRemove: string,
-    fieldSetToAdd: string
+    fieldSetToAdd: string,
+    indexPos: number = 0
   ): ExplorerFields => {
     const nextFields = { ...fieldState };
+
+    // find and remove field
     nextFields[fieldSetToRemove] = nextFields[fieldSetToRemove].filter(
       (fd: IField) => fd.name !== field.name
     );
-    nextFields[fieldSetToAdd] = [...nextFields[fieldSetToAdd], field];
+
+    // add field at specified index, will resolve to 0 if not specified
+    const addedFieldSet = [...nextFields[fieldSetToAdd]]; // copies value, not reference
+    addedFieldSet.splice(indexPos, 0, field);
+    nextFields[fieldSetToAdd] = addedFieldSet;
+
     return nextFields;
   };
 
-  const updateStoreFields = (fieldsData: ExplorerFields, tabID: string, modifiedField: string) => {
-    batch(() => {
-      dispatch(
-        updateFields({
-          tabId: tabID,
-          data: {
-            ...fieldsData,
-          },
-        })
-      );
-      dispatch(
-        sortFields({
-          tabId: tabID,
-          data: [modifiedField],
-        })
-      );
-    });
+  const updateStoreFields = (fieldsData: ExplorerFields) => {
```

Comment:
We need useCalkback here is it? As I'm seeing you add this as part of dependency of another useCallback

Reply:
added